workflows/detaxizer.nf

```diff
@@ -3,53 +3,67 @@
+include { GENERATE_DOWNSTREAM_SAMPLESHEETS                          } from '../subworkflows/local/generate_downstream_samplesheets/main.nf'
+
+include { FASTP                                                     } from '../modules/nf-core/fastp/main'
+include { KRAKEN2_KRAKEN2 as KRAKEN2_KRAKEN2                        } from '../modules/nf-core/kraken2/kraken2/main'
+include { KRAKEN2_KRAKEN2 as KRAKEN2_POST_CLASSIFICATION_FILTERED   } from '../modules/nf-core/kraken2/kraken2/main'
+include { KRAKEN2_KRAKEN2 as KRAKEN2_POST_CLASSIFICATION_REMOVED    } from '../modules/nf-core/kraken2/kraken2/main'
+include { BBMAP_BBDUK                                               } from '../modules/nf-core/bbmap/bbduk/main'
+include { BLAST_BLASTN                                              } from '../modules/nf-core/blast/blastn/main'
+include { BLAST_MAKEBLASTDB                                         } from '../modules/nf-core/blast/makeblastdb/main'
+
+include { RENAME_FASTQ_HEADERS_PRE                                  } from '../modules/local/rename_fastq_headers_pre'
+include { KRAKEN2PREPARATION                                        } from '../modules/local/kraken2preparation'
+include { PARSE_KRAKEN2REPORT                                       } from '../modules/local/parse_kraken2report'
+include { ISOLATE_KRAKEN2_IDS                                       } from '../modules/local/isolate_kraken2_ids'
+include { ISOLATE_BBDUK_IDS                                         } from '../modules/local/isolate_bbduk_ids'
+include { MERGE_IDS                                                 } from '../modules/local/merge_ids'
+include { PREPARE_FASTA4BLASTN                                      } from '../modules/local/prepare_fasta4blastn'
+include { FILTER_BLASTN_IDENTCOV                                    } from '../modules/local/filter_blastn_identcov'
+include { FILTER                                                    } from '../modules/local/filter'
+include { RENAME_FASTQ_HEADERS_AFTER                                } from '../modules/local/rename_fastq_headers_after'
+include { SUMMARY_CLASSIFICATION                                    } from '../modules/local/summary_classification'
+include { SUMMARY_BLASTN                                            } from '../modules/local/summary_blastn'
+include { SUMMARIZER                                                } from '../modules/local/summarizer'
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     RUN MAIN WORKFLOW
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-// speficy the fasta channel if it is not provided via --fasta
-def fasta = Channel.empty()
+// speficy the fasta_blastn channel if it is not provided via --fasta_blastn
+def fasta_blastn = Channel.empty()
+
+if ( !params.fasta_blastn && params.validation_blastn ) {
+    fasta_blastn = Channel.fromPath(getGenomeAttribute('fasta'))
```

Comment:
```suggestion
    ch_fasta_blastn = Channel.fromPath(getGenomeAttribute('fasta'))
```

Reply:
The convention is to prefix _any_ channel object with `ch_`, I would do this for all these input channel preparations as you've used the same convention later in the script